### PR TITLE
Fix deprecation warnings - Update to React 15.6.2

### PR DESCRIPTION
**Pull request by abarriel**
`React.PropTypes`, `react-addons-test-utils`, and `react-addons-pure-render-mixin` are depracted.
To avoid any future complications, here are the new changes:

- Removed `React.PropTypes` for `PropTypes`, a separate package,
use a codemods from `react-codemods` lib for applying those changes.
Unique name of PropTypes object imported to have a generics code over
the lib ( removed  { T } in favor to { PropTypes } )

- Removed `react-addons-test-utils` deprecated since 15.5 after 15.5 react-dom/utils can be used,
for shallow rendering react-test-render is use since after 15.5

- Update Package `React` and `React Dom`

- Removed `react-addons-pure-render-mixin`, we not longer use this addons,
 `React.pureComponent` provide exactly the same shouldComponentUpdate(with a shallowCompare) as
  PureRender did before. To match these changes along the lib, we add
  a new comparator ( which return true  if the component is a React.Component
  or React.PureComponent ), because in the previous version react-nexus
  didn't have any information about PureComponent imported from React.

- Remove should from dependencies, it was imported twice on in devdependencies and dependencies

### CODE REVIEW

File: lib/__tests__/node/components.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import React, { Component, PropTypes as T } from 'react';
+import T from 'prop-types';
```
Comment (eTorAken) [inline lib/__tests__/node/components.js]: hésite pas à tout de la même façon : `PropTypes` ou `T` mais pas les deux ;)

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,8610 @@
+{
```
Comment (eTorAken) [inline package-lock.json]: Could you add this file in a separate commit please? I don't think it has something to do with React 15.6 update.

**Comment by eTorAken** [comment]
Also "Remove should from dependencies" in your commit message. Please explain why.

File: package-lock.json (hunk 2 above)
Comment (abarriel) [reply package-lock.json]: this package-lock.json locks the new version of the fresh packages added (React, ReactDom, React-test-renderer)

**Comment by abarriel** [comment]
This PR should be clean. I apply some changes

File: lib/actions.js [hunk 3]
```diff
@@ -12,35 +12,24 @@ import defaultFluxKey from './defaultFluxKey';
  * @param {Object} options Options object.
  * @param {String} [options.displayName] The displayName of the wrapper component (useful for debugging).
  * @param {String} [options.fluxKey] Use a specific string as fluxKey rather than the default one.
- * @param {Function} [options.shouldNexusComponentUpdate] Use specific function for shouldComponentUpdate event.
  * @return {Component} Enhanced component.
  */
 function actions(getBindings, {
     displayName = void 0,
     fluxKey = defaultFluxKey,
-    shouldNexusComponentUpdate = PureRenderMixin.shouldComponentUpdate,
```
Comment (antca) [inline lib/actions.js]: We need to keep this option.

File: lib/util/isReactCompositeComponent.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export default function isReactCompositeComponent(component) {
+  if (typeof component.prototype === 'object') {
+    if (component.prototype.isReactComponent || component.prototype.isPureReactComponent) {
```
Comment (antca) [inline lib/util/isReactCompositeComponent.js]: `|| component.prototype.isPureReactComponent` is not required here.

**Pull request merged**